_security/access-control/authentication-tokens.md

```diff
@@ -0,0 +1,123 @@
+		* Role Security Mode On (default value) - Roles claim will be encrypted.
+			* Encrypted mapped roles (`er`)
+		* Role Security Mode Off - Roles claims will be in plain-text, and both mapped roles and backend roles will be included in the claim [[related discussion](https://github.com/opensearch-project/security/issues/2865)]
+			* Decrypted mapped roles in plain text (`dr`)
+			* Decrypted backend roles (`br`)
+
+The OpenSearch Security plugin will be responsible for handling encryption and decryption processes. This approach ensures the protection of user information, even when traversing the trust boundary between OpenSearch and any third party services.
+
+### 3.0 API endpoint
+There will be an new API endpoint `POST /_plugins/_security/api/generateonbehalfoftoken` on the security plugin that will allow users to create a short-lived self-issued on-behalf-of token to perform certain actions on behalf of a user.
+
+To access this API endpoint, the request body should contain three API parameters:
+
+* description: This allows the use to articulate the purpose for requesting this token, providing clarity and transparency.
+* service (optional): This parameter is directed to the audience claim of the on-behalf-of token. It offers users the opportunity to designate the target service for which they intend to use the token. Although this is an optional parameter, if not specified, the default value is set to "self-issued".
+* durationSeconds (optional): This parameter allows users to customize the token's expiration time according to its anticipated usage. However, the maximum duration is capped at 600 seconds to maintain security. If not specified, the default duration is set to 300 seconds.
+* Here is an example of requesting an on-behalf-of token with lifespan of 3 mins as user '“admin” for testing purpose:
+```
+curl -XPOST https://localhost:9200/_plugins/_security/api/generateonbehalfoftoken -u 'admin:admin' -H 'Content-Type: application/json' --data
+'{ 
+   "description":"Testing",
+   "service":"Testing Service",
+   "durationSeconds":"180"
+}'
+```
+
+### 4.0 Additional Authorization Restriction ([related discussion](https://github.com/opensearch-project/security/issues/2891))
+
+While the conversation about the usage of on-behalf-of (OBO) tokens continues, it is critical to manage certain edge cases. Even though an OBO token can act as a valid Bearer authorization header for any API access, there needs to be certain limitations. For instance, it should be forbidden to use an OBO token to access the API endpoint to issue another OBO token. Similarly, using an OBO token to access the reset password API in order to modify a user's authentication information should be disallowed. These preventive measures are necessary to uphold the integrity and security of the system.
+
+## Service Accounts
+
```

Comment:
Could you insert an intro sentence here? Per our style guide, we cannot have one heading immediately following the other.